spring-web/src/main/java/org/springframework/http/HttpHeaders.java

```diff
@@ -397,30 +396,31 @@
 	private static final DateTimeFormatter[] DATE_FORMATTERS = new DateTimeFormatter[] {
 			DateTimeFormatter.RFC_1123_DATE_TIME,
 			DateTimeFormatter.ofPattern("EEEE, dd-MMM-yy HH:mm:ss zz", Locale.US),
-			DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss yyyy",Locale.US).withZone(GMT)
+			DateTimeFormatter.ofPattern("EEE MMM dd HH:mm:ss yyyy", Locale.US).withZone(GMT)
 	};
 
 
-	private final Map<String, List<String>> headers;
+	private final MultiValueMap<String, String> headers;
 
 	private final boolean readOnly;
 
+	@Nullable
+	private MediaType cachedContentType;
+
 
 	/**
-	 * Constructs a new, empty instance of the {@code HttpHeaders} object.
+	 * Construct a new, empty instance of the {@code HttpHeaders} object.
 	 */
 	public HttpHeaders() {
-		this(new LinkedCaseInsensitiveMap<>(8, Locale.ENGLISH), false);
+		this(new DefaultHttpHeadersMap(), false);
 	}
 
 	/**
-	 * Private constructor that can create read-only {@code HttpHeader} instances.
+	 * Construct a new {@code HttpHeaders} instance backed by an existing map.
 	 */
-	private HttpHeaders(Map<String, List<String>> headers, boolean readOnly) {
+	private HttpHeaders(MultiValueMap<String, String> headers, boolean readOnly) {
 		if (readOnly) {
-			Map<String, List<String>> map = new LinkedCaseInsensitiveMap<>(headers.size(), Locale.ENGLISH);
-			headers.forEach((key, valueList) -> map.put(key, Collections.unmodifiableList(valueList)));
-			this.headers = Collections.unmodifiableMap(map);
+			this.headers = new UnmodifiableMultiValueMap<>(headers);
```

Comment:
I think we should create a `ReadOnlyHttpHeades` instead of using a `UnmodifiableMultiValueMap` here, see below.